src/components/util/colorUtils.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+const BASE_16 = 16;
+
+const calculateColors = function(elements) {
```

Comment:
Please read up on D3's `color scales`. You don't have to do this :D

Example: https://jsfiddle.net/vpf5t33z/